applications/crossbar/src/modules/cb_recordings.erl

```diff
@@ -0,0 +1,189 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2012-2016, 2600Hz INC
+%%% @doc
+%%%
+%%% Provides access to stored call recordings:
+%%%
+%%% To play in browser:
+%%% https://my_crossbar.tld:8443/v1/accounts/33ca...1452/recordings/201412-6471-8c79ae@192.1.1.2/attachment?inline=true&auth_token=05c1...7a075
+%%%
+%%% To download:
+%%% https://my_crossbar.tld:8443/v1/accounts/33ca...1452/recordings/201412-6471-8c79ae@192.1.1.2/attachment?auth_token=05c1...7a075
+%%%
+%%% @end
+%%% @contributors:
+%%%
+%%%   Original pull request https://github.com/2600hz/kazoo/pull/1218
+%%%     OnNet (Kirill Sysoev github.com/onnet)
+%%%     Dinkor (Andrew Korniliv github.com/dinkor)
+%%%
+%%%   refactor
+%%%     lazedo (Luis Azedo github.com/2600hz)
+%%%
+%%%-------------------------------------------------------------------
+-module(cb_recordings).
+
+-export([init/0
+         ,allowed_methods/0, allowed_methods/1, allowed_methods/2
+         ,resource_exists/0, resource_exists/1, resource_exists/2
+         ,content_types_provided/3
+         ,validate/1, validate/2, validate/3
+        ]).
+
+-include("../crossbar.hrl").
+
+-define(CB_LIST, <<"recordings/crossbar_listing">>).
+-define(CB_LIST_BY_OWNERID, <<"recordings/listing_by_user">>).
+
+-define(ATTACHMENT, <<"attachment">>).
+-define(INLINE, <<"inline">>).
+-define(MEDIA_MIME_TYPES, [{<<"audio">>, <<"mpeg">>}]).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Initializes the bindings this module will respond to.
+%% @end
+%%--------------------------------------------------------------------
+-spec init() -> 'ok'.
+init() ->
+    _ = crossbar_bindings:bind(<<"*.allowed_methods.recordings">>, ?MODULE, 'allowed_methods'),
+    _ = crossbar_bindings:bind(<<"*.resource_exists.recordings">>, ?MODULE, 'resource_exists'),
+    _ = crossbar_bindings:bind(<<"*.content_types_provided.recordings">>, ?MODULE, 'content_types_provided'),
+    crossbar_bindings:bind(<<"*.validate.recordings">>, ?MODULE, 'validate').
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Given the path tokens related to this module, what HTTP methods are
+%% going to be responded to.
+%% @end
+%%--------------------------------------------------------------------
+-spec allowed_methods() -> http_methods().
+-spec allowed_methods(path_token()) -> http_methods().
+-spec allowed_methods(path_token(), path_token()) -> http_methods().
+
+allowed_methods() -> [?HTTP_GET].
+allowed_methods(_Id ) -> [?HTTP_GET].
```

Comment:
Can you name these `_RecordingId`? The auto-gen tool will look nicer with `{RECORDING_ID}` in the path that just `{ID}`.